pgrep.c

```diff
@@ -233,3 +293,63 @@ static void *run_signal_thread(void *arg) {
 
     return NULL;
 }
+
+static int install_sigusr2_handler(void) {
+    struct sigaction sa;
+
+    memset(&sa, 0, sizeof(struct sigaction));
+    sa.sa_handler = handle_sigusr2;
+    sigaction(SIGUSR2, &sa, NULL);
+
+    return PHPSPY_OK;
+}
+
+/* Copied from linux: tools/perf/util/time-utils.c */
+static int fetch_current_timestamp(char *buf, size_t sz) {
+    struct timeval tv;
+    struct tm tm;
+    char dt[32];
+
+    if (gettimeofday(&tv, NULL) || !localtime_r(&tv.tv_sec, &tm))
+        return -1;
+
+    if (!strftime(dt, sizeof(dt), "%Y%m%d%H%M%S", &tm))
+        return -1;
+
+    snprintf(buf, sz, "%s%02u", dt, (unsigned)tv.tv_usec / 10000);
+
+    return 0;
+}
+
+static void handle_sigusr2(int signal) {
+#define TIMESTAMP_BUF_SIZE (128)
+    char timestamp_buf[TIMESTAMP_BUF_SIZE];
+    char new_path[PATH_MAX];
+    int tmp_output_fd = -1;
+
+    (void)signal;
+    if (strcmp(opt_path_output, STDOUT_OUTPUT) == 0) {
+        log_error("Not rotating output file in stdout output mode\n");
+        return;
+    }
+
+    /* See lock explanation comment above the lock declaration  */
+    pthread_rwlock_wrlock(&pgrep_mode_rw_lock);
```

Comment:
shouldn't do any of this in the signal handler itself because it's unsafe `signal-safety(7)`. need to set a flag and check for it in the loop

Reply:
done